### PR TITLE
Expose prior in gen kw config

### PR DESCRIPTION
**Pull request by ManInFez**

### CODE REVIEW

File: python/res/enkf/key_manager.py [hunk 1]
```diff
@@ -177,3 +177,15 @@ def isGenDataKey(self, key):
...
+        gen_kw_keys = self._ert().ensembleConfig().getKeylistFromImplType(ErtImplType.GEN_KW)
+        gen_kw_keys = [key for key in gen_kw_keys]
+
+        all_gen_kw_priors = {}
+        for key in gen_kw_keys:
+            enkf_config_node = self._ert().ensembleConfig().getNode(key)
+            gen_kw_config = enkf_config_node.getModelConfig()
+            all_gen_kw_priors[key] = gen_kw_config.get_priors()
+
+        return all_gen_kw_priors
```
Comment (markusdregi) [inline python/res/enkf/key_manager.py]: Missing new-line

File: python/res/enkf/config/gen_kw_config.py [hunk 2]
```diff
@@ -113,3 +116,32 @@ def __eq__(self, other):
...
+            parameter_values = self._get_function_parameter_values(i)
+            el = {
+                "key" : key,
+                "function" : function_type,
+                "parameters" : {
+                    name : value for (name, value) in zip(parameter_names, parameter_values)
+                }
+            }
+            priors.append(el)
+        return priors
```
Comment (markusdregi) [inline python/res/enkf/config/gen_kw_config.py]: Missing new-line

File: lib/enkf/trans_func.cpp [hunk 3]
```diff
@@ -348,3 +348,13 @@ double trans_func_eval( const trans_func_type * trans_func , double x) {
...
+
+stringlist_type * trans_func_get_param_names(const trans_func_type * trans_func){
+  return trans_func->param_names;
+}
+double_vector_type * trans_func_get_params(const trans_func_type * trans_func){
+  return trans_func->params;
+}
+const char * trans_func_get_name(const trans_func_type * trans_func){
+  return trans_func->name;
+}
```
Comment (markusdregi) [inline lib/enkf/trans_func.cpp]: Missing new-line

File: python/res/enkf/key_manager.py [hunk 4]
```diff
@@ -177,3 +177,15 @@ def isGenDataKey(self, key):
     def isMisfitKey(self, key):
         """ :rtype: bool """
         return key in self.misfitKeys()
+
+    def genKwPriors(self):
```
Comment (markusdregi) [inline python/res/enkf/key_manager.py]: I suggest we make this one camel case.

File: python/res/enkf/config/gen_kw_config.py [hunk 5]
```diff
@@ -35,6 +35,9 @@ class GenKwConfig(BaseCClass):
     _size                 = ResPrototype("int   gen_kw_config_get_data_size(gen_kw_config)")
     _iget_name            = ResPrototype("char* gen_kw_config_iget_name(gen_kw_config, int)")
```
Comment (markusdregi) [inline python/res/enkf/config/gen_kw_config.py]: No need for an empty line here, nor in the other files like headers etc.

File: python/tests/res/enkf/data/test_gen_kw_config.py [hunk 6]
```diff
@@ -34,3 +34,23 @@ def test_gen_kw_config(self):
...
+            with open("template.txt","w") as f:
+                f.write("Hello")
+
+            with open("parameters.txt","w") as f:
+                f.write("KEY1  UNIFORM 0 1\n")
+                f.write("KEY2  UNIFORM 0 1\n")
+                f.write("KEY3  UNIFORM 0 1\n")
+
+            
+            parameter_file = "parameters.txt"
```
Comment (markusdregi) [inline python/tests/res/enkf/data/test_gen_kw_config.py]: Please define these at the top and use the variables when writing the files.

File: python/res/enkf/key_manager.py (hunk 4 above)
Comment (markusdregi) [reply python/res/enkf/key_manager.py]: In addition, this function is in dire need of a test or two ;)

**Comment by markusdregi** [comment]
Looks good 👍 Please address the comments...

File: python/tests/res/enkf/data/test_gen_kw_config.py [hunk 7]
```diff
@@ -34,3 +34,23 @@ def test_gen_kw_config(self):
             conf = GenKwConfig("KEY", template_file, parameter_file)
             conf = GenKwConfig("KEY", template_file, parameter_file_comments)
             self.assertEqual(len(conf), 3)
+    
+    
+    def test_gen_kw_config_get_priors(self):
+        with TestAreaContext("gen_kw_config"):
```
Comment (markusdregi) [inline python/tests/res/enkf/data/test_gen_kw_config.py]: Also, the `tmpdir` construct from `tests.utils` is preferred unless `TestAreaContext` is truly necessary.

File: python/res/enkf/key_manager.py (hunk 1 above)
Comment (ManInFez) [reply python/res/enkf/key_manager.py]: 👍

File: python/res/enkf/config/gen_kw_config.py (hunk 2 above)
Comment (ManInFez) [reply python/res/enkf/config/gen_kw_config.py]: 👍

File: python/res/enkf/key_manager.py (hunk 4 above)
Comment (ManInFez) [reply python/res/enkf/key_manager.py]: is it not the same format as all the other functions in this file?

**Comment by ManInFez** [comment]
Test this please!

**Comment by ManInFez** [comment]
Test this please!

**Comment by markusdregi** [comment]
It would be beneficiary to add tests verifying that all the different distributions are served correctly.

File: python/res/enkf/config/gen_kw_config.py [hunk 8]
```diff
@@ -113,3 +115,32 @@ def __eq__(self, other):
             return False
 
         return True
+
+    def getPriors(self):
```
Comment (markusdregi) [inline python/res/enkf/config/gen_kw_config.py]: Lets keep new functions `snake_case`™️ Sorry for the confusing messages here, but the previous comment was a "typo"

Comment (ManInFez) [reply python/res/enkf/config/gen_kw_config.py]: Ah missed that one

File: python/tests/res/enkf/data/test_gen_kw_config.py [hunk 9]
```diff
@@ -34,3 +34,90 @@ def test_gen_kw_config(self):
...
+            f.write("KEY8  DERRF 0 1 2 3 4\n")
+            f.write("KEY9  LOGUNIF 0 1\n")
+            f.write("KEY10  CONST 10\n")
+            
+
+        conf = GenKwConfig("KEY", template_file, parameter_file)
+        priors = conf.get_priors()
+        self.assertEqual(len(conf), 10)
+
+        print(priors[7])
```
Comment (markusdregi) [inline python/tests/res/enkf/data/test_gen_kw_config.py]: Please remove this one ;)

**Review by markusdregi — APPROVED** [review]
Remove the print statement and get the tests to pass. Then this is good to go 👍

**Comment by pinkwah** [comment]
Jenkins test this please

**Pull request merged**